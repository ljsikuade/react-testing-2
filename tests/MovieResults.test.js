import React from "react";
import MovieResults from "../src/components/MovieResults";
import renderer from "react-test-renderer";

const movieArr = [
  {
    imdbID: "bb",
    Year: 2015,
    Title: "Yo",
    Poster: "xx"
  },
  {
    imdbID: "adfs",
    Year: 2017,
    Title: "We love testing",
    Poster: "fkjn"
  },
  {
    imdbID: "aa",
    Year: 2019,
    Title: "CDXF",
    Poster: "faslfdk"
  }
];

describe("Results", () => {
  test("Array is correct", () => {
    const movieComp = renderer
      .create(<MovieResults movies={movieArr} />)
      .toJSON();
    expect(movieComp).toMatchSnapshot();
  });
});
