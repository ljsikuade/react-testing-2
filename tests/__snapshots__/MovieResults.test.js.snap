// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Array is correct 1`] = `
<div>
  <ul>
    <li>
      <h2>
        Yo
         - 
        2015
      </h2>
      <div>
        <span>
          Movie has 
          0
           likes
        </span>
        <button
          onClick={[Function]}
        >
          Like
        </button>
      </div>
      <img
        src="xx"
      />
    </li>
    <li>
      <h2>
        We love testing
         - 
        2017
      </h2>
      <div>
        <span>
          Movie has 
          0
           likes
        </span>
        <button
          onClick={[Function]}
        >
          Like
        </button>
      </div>
      <img
        src="fkjn"
      />
    </li>
    <li>
      <h2>
        CDXF
         - 
        2019
      </h2>
      <div>
        <span>
          Movie has 
          0
           likes
        </span>
        <button
          onClick={[Function]}
        >
          Like
        </button>
      </div>
      <img
        src="faslfdk"
      />
    </li>
  </ul>
</div>
`;
